libs/user-identity-service/src/main/java/uk/gov/di/ipv/core/library/service/UserIdentityService.java

```diff
@@ -150,11 +159,49 @@ public UserIdentity generateUserIdentity(String userId, String sub, String vot)
             Optional<JsonNode> drivingPermitClaim =
                     generateDrivingPermitClaim(successfulVCStoreItems);
             drivingPermitClaim.ifPresent(userIdentityBuilder::drivingPermitClaim);
+            if (configService.enabled(EXIT_CODES)) {
+                userIdentityBuilder.exitCode(toAlwaysRequiredExitCode(contraIndicators));
+            }
+        } else {
+            if (configService.enabled(EXIT_CODES)
+                    && contraIndicators.getContraIndicatorScore(
+                                    configService.getContraIndicatorConfigMap())
+                            >= Integer.parseInt(
+                                    configService.getSsmParameter(CI_SCORING_THRESHOLD))) {
+                userIdentityBuilder.exitCode(toExitCode(contraIndicators));
```

Comment:
Interested to know what you think, but having these methods here feels cleaner to me :)

A minor point, but looking forward to the next ticket, maybe we should extract the breaching check as well, so we have two methods for `getSuccessExitCode` and `getFailedExitCode`, rather than being CI-specific? Happy for you to work it out there instead if you prefer.

Reply:
Yeah, I agree this feels a lot cleaner. No need for the `ContraIndicators` class to know anything about exit codes. Less faff with calling config service and passing stuff through.

Not sure I follow on the breaching check - one for a face to face chat? Will get this in though and address in the next ticket I think.